build-tools/packages/build-cli/src/library/release.ts

```diff
@@ -70,30 +72,59 @@ export interface ReleaseRanges {
 	 * A tilde version range. Equivalent to patch.
 	 */
 	tilde: string;
+
+	/**
+	 * A legacy compat range
+	 */
+	legacyCompat: string;
 }
 
 /**
  * Get the release ranges for a version string.
  *
  * @param version - The version.
+ * @param releaseGroup - The release group to filter by
+ * @param context - The repo {@link Context}. The context is retrieved and cached the first time this method is called.
+ * @param compatVersionInterval - The multiple of minor versions to use for calculating the next version in the range.
  * @param scheme - If provided, this version scheme will be used. Otherwise the scheme will be detected from the
  * version.
  * @returns The {@link ReleaseRanges} for a version string
  */
-export const getRanges = (version: ReleaseVersion, scheme?: VersionScheme): ReleaseRanges => {
+export const getRanges = (
```

Comment:
I don't think you should change this function. A caller shouldn't need to pass all these extra properties just to generate ranges. It seems like you want this to be an alternative range representation but only for some versions. That logic should live in one place and as close to the report output code as possible. I would revert this and put the logic only in the toReportKind function. The main reason is that this is NOT a new version scheme - it's an alternative range representation for any input version. The only code that needs to treat packages "special" is the report generation code.

In other words, it's OK if you pass this a server package version (e.g. 5.0). This function will return `>=5.0.0 <= 5.10.0`, but that's fine - you can adjust the server values at the last second in the toReportKind function.

Reply:
Would this need to be handled in both the full report and the legacy-compat report? The full report includes incorrect ranges, and if the Loop bump tooling is updated to use just the full report in the future, the server ranges will be incorrect.

Reply:
This is why I think we need to be explicit about whether this is a new version scheme or just an alternative range representation. If it's the latter, then the right thing is for the full report to contain all the ranges, and the consumer of the full report can choose which range to use for each package. That logic can even be based on the release group because that's in the full report too. If we can change the loop logic, then we can reduce this change to just the new range in the full report, and skip the new report kind.

Reply:
But if the logic needs to live on our side, then it can live in toReportKind and follow the same logic that loop would use.

Reply:
I think it's a range representation and then the full report can include the server range too (>=5.0.0 <5.10.0). 

fyi, @jason-ha let me know if you see it differently.